attest/api/src/convert/quote3.rs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2023 The MobileCoin Foundation
+
+//! Convert to/from attest::Quote3
+
+use crate::{attest, ConversionError};
+use mc_sgx_dcap_types::Quote3;
+
+impl<T: AsRef<[u8]>> From<&Quote3<T>> for attest::Quote3 {
+    fn from(src: &Quote3<T>) -> Self {
+        let mut dst = Self::new();
+        dst.set_data(src.as_ref().to_vec());
+
+        dst
+    }
+}
```

Comment:
Since quote as as_ref and can convert directly from bytes, I figured doing it directly instead of streaming like `Collateral` does would be fine and the quote is not likely to change